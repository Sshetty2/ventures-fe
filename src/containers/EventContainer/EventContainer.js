import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import EventCard from '../../components/EventCard/EventCard';

export const EventContainer = (props) => {
    return (
      <div className='EventContainer'>
      {
        props.events.map(event => 
          <EventCard 
            id={event.id}
            name={event.attributes.name}
            image={event.attributes.image_url}
            city={event.attributes.city}
            state={event.attributes.state}
            date={event.attributes.start_date}
          />
        )
      }
      </div>
    )
}

export const mapStateToProps = (state) => ({
  events: state.events,
})

export default connect(mapStateToProps)(EventContainer);

