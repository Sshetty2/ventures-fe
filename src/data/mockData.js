export const mockEvents = {
  data: [
    {
      id: "1",
      type: "event",
      attributes: {
        name: "Shelf Road Craggin' Classic",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "2",
      type: "event",
      attributes: {
        name: "Hueco Rock Rodeo",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "3",
      type: "event",
      attributes: {
        name: "Flash Foxy Women's Climbing Festival",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "4",
      type: "event",
      attributes: {
        name: "Flash Foxy Summerfest",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "5",
      type: "event",
      attributes: {
        name: "International Climber's Festival",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "6",
      type: "event",
      attributes: {
        name: "Leavenworth Rock Fest",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    },
    {
      id: "3",
      type: "event",
      attributes: {
        name: "Joes Valley Fest",
        city: "Canon Springs",
        state: "CO",
        event_type: "climbing",
        price: 35,
        start_date: "10-19-2018",
        end_date: "10-21-2018",
        description:
          "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
        event_url: "https://americanalpineclub.org/shelf-road-cc",
        image_url:
          "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
        video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo"
      }
    }
  ]
};

export const mockUser = {
  name: 'Jane Doe',
  image_url: 'https://alumni.turing.io/sites/default/files/styles/graduate_full_profile/public/squareheadshot1.jpg?itok=EGlR-8YR',
  bio: 'Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!!'
}

export const mockUserEvents = [
  {
    id: "6",
    type: "event",
    attributes: {
      name: "Leavenworth Rock Fest",
      city: "Canon Springs",
      state: "CO",
      event_type: "climbing",
      price: 35,
      start_date: "10-19-2018",
      end_date: "10-21-2018",
      description:
        "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
      event_url: "https://americanalpineclub.org/shelf-road-cc",
      image_url:
        "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
      video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo",
      status: 'attending'
    }
  },
  {
    id: "3",
    type: "event",
    attributes: {
      name: "Joes Valley Fest",
      city: "Canon Springs",
      state: "CO",
      event_type: "climbing",
      price: 35,
      start_date: "10-19-2018",
      end_date: "10-21-2018",
      description:
        "This October 19-21, head on down to the best limestone sport cragging Colorado has to offer, and with the American Alpine Club and CAMP USA, and be treated to a weekend of climbing, community, clinics, contests, beer, slideshows, films, giveaways, stewardship and fun.",
      event_url: "https://americanalpineclub.org/shelf-road-cc",
      image_url:
        "https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg",
      video_url: "https://www.youtube.com/embed/Cnjv_Brs2Bo",
      status: 'wishlist'
    }
  }
]