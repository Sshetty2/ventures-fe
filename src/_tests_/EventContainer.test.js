import React from 'react';
import { shallow } from 'enzyme';
import { EventContainer, mapStateToProps } from '../containers/EventContainer/EventContainer';
import { mockEvents, mockUserEvents } from '../data/mockData';

const mockProps = {
  events: mockEvents.data,
  pathname: '/profile',
  searchText: '',
  userEvents: mockUserEvents,
};

describe('EventContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<EventContainer {...mockProps} />)
  });

  it('should match the snapshot for the /profile path', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot for the /home path with no search text', () => {
    wrapper = shallow(<EventContainer events={mockEvents.data} pathname={'/home'} searchText={mockProps.searchText} />)
    expect(wrapper).toMatchSnapshot()
  });

  it('should match the snapshot for the /home path with search text', () => {
    wrapper = shallow(<EventContainer events={mockEvents.data} pathname={'/home'} searchText='sea' />)
    expect(wrapper).toMatchSnapshot()
  });

  describe('mapStateToProps', () => {
    it('should return a props object with events', () => {
      const mockState = {
        events: mockEvents,
        searchText: mockProps.searchText,
        userEvents: mockProps.mockUserEvents,
        isLoading: true,
        error: ''
      };
      const expectedState = {
        events: mockEvents,
        searchText: mockProps.searchText,
        userEvents: mockProps.mockUserEvents
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    });
  })
})