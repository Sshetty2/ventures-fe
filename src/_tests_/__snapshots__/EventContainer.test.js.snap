// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventContainer should match the snapshot for the /home path 1`] = `
<div
  className="EventContainer home"
>
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="1"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="1"
    name="Shelf Road Craggin' Classic"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="2"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="2"
    name="Hueco Rock Rodeo"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="3"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="3"
    name="Flash Foxy Women's Climbing Festival"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="4"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="4"
    name="Flash Foxy Summerfest"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="5"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="5"
    name="International Climber's Festival"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="6"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="6"
    name="Leavenworth Rock Fest"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="3"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="3"
    name="Joes Valley Fest"
    pathname="/home"
    state="CO"
  />
</div>
`;

exports[`EventContainer should match the snapshot for the /home path with no search text 1`] = `
<div
  className="EventContainer home"
>
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="1"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="1"
    name="Shelf Road Craggin' Classic"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="2"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="2"
    name="Hueco Rock Rodeo"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="3"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="3"
    name="Flash Foxy Women's Climbing Festival"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="4"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="4"
    name="Flash Foxy Summerfest"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="5"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="5"
    name="International Climber's Festival"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="6"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="6"
    name="Leavenworth Rock Fest"
    pathname="/home"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="3"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="3"
    name="Joes Valley Fest"
    pathname="/home"
    state="CO"
  />
</div>
`;

exports[`EventContainer should match the snapshot for the /home path with search text 1`] = `
<div
  className="EventContainer home"
>
  <h2>
    No events to show
  </h2>
</div>
`;

exports[`EventContainer should match the snapshot for the /profile path 1`] = `
<div
  className="EventContainer profile"
>
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="1"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="1"
    name="Shelf Road Craggin' Classic"
    pathname="/profile"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="2"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="2"
    name="Hueco Rock Rodeo"
    pathname="/profile"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="3"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="3"
    name="Flash Foxy Women's Climbing Festival"
    pathname="/profile"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="4"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="4"
    name="Flash Foxy Summerfest"
    pathname="/profile"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="5"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="5"
    name="International Climber's Festival"
    pathname="/profile"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="6"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="6"
    name="Leavenworth Rock Fest"
    pathname="/profile"
    state="CO"
  />
  <EventCard
    city="Canon Springs"
    date="10-19-2018"
    id="3"
    image="https://static1.squarespace.com/static/55830fd9e4b0ec758c892f81/57bc7284f7e0abe0efbb523e/59417dc21b631bad37b19a74/1497464437416/_MG_1604.jpg"
    key="3"
    name="Joes Valley Fest"
    pathname="/profile"
    state="CO"
  />
</div>
`;
